app/utils/language.js

```diff
@@ -0,0 +1,13 @@
+import config from '../config/environment';
+import {escapeRegex} from './string';
+
+/**
+ * @param {Object} request - FastBoot request
+ * @returns {string|null}
+ */
+export default function getLanguageCodeFromRequest(request = null) {
+	const path = request ? request.get('path') : window.location.pathname,
+		matches = path.match(/^(\/[a-z]{2,3}(?:-[a-z-]{2,12})?)/);
```

Comment:
This pattern is used in 3 places. Can we reuse it using e.g. `new RegExp(langPathPart + 'the rest of pattern')`?